src/test/java/com/chikli/hudson/plugin/naginator/NaginatorPublisherTest.java

```diff
@@ -267,6 +271,155 @@ public void testRegexpForMatrixChild() throws Exception {
     }
     
     /**
+     * Tests the behavior when <code>regexpForMatrixStrategy</code> is <code>TestChildrenRetriggerMatched</code>
+     * and <code>retunMatrixPart</code> is <code>false</code>.
+     * 
+     * @throws Exception
+     */
+    @Test
+    public void testRegexpForMatrixChildWithoutMatrixPart() throws Exception {
+        MatrixProject p = j.createMatrixProject();
+        AxisList axes = new AxisList(
+                new Axis("axis1", "value1", "value2", "value3")
+        );
+        p.setAxes(axes);
+        
+        // all axis fails, and outputs "I am (axis value)"
+        p.getBuildersList().add(new MyBuilder("I am ${axis1}", Result.FAILURE));
+        NaginatorPublisher naginator = new NaginatorPublisher(
+                "thatcannotbe",
+                false,  // rerunIfUnstable
+                false,  // retunMatrixPart
+                true,   // checkRegexp
+                1,      // maxSchedule
+                new FixedDelay(0)
+        );
+        naginator.setRegexpForMatrixStrategy(RegexpForMatrixStrategy.TestChildrenRetriggerMatched);
+        naginator.setNoChildStrategy(NoChildStrategy.DontRun);
+        p.getPublishersList().add(naginator);
+        
+        p.scheduleBuild2(0);
+        j.waitUntilNoActivity();
+        
+        // build is rescheduled
+        // as regular expression is not applied actually.
+        assertEquals(2, p.getLastBuild().number);
+    }
+    
+    /**
+     * Tests the behavior when <code>regexpForMatrixStrategy</code> is <code>TestChildrenRetriggerAll</code>.
+     * 
+     * @throws Exception
+     */
+    @Test
+    public void testRegexpForMatrixChildAndRetriggerAll() throws Exception {
```

Comment:
This is a case described by @mjainta .
axis1=value1 and axis1=value3 failed, regexp matched axis1=value1, and retriggers both axis1=value1 and axis1=value3.